npx nx@latest init --integrated on a fresh angular app does nothing

### Current Behavior

I created a fresh angular 19 app running
`ng new`

Then ran `npx nx@latest init --integrated` as per the documentation and simply got this silent failure.

![Image](https://github.com/user-attachments/assets/8a0786cc-0bdd-4ec6-a0dd-d5611d12a755)

### Expected Behavior

I'd expect the migration to correctly set up a nx workspace.

### GitHub Repo

https://github.com/alvaromartmart/nx-31291-repro

### Steps to Reproduce

1. Create fresh angular app with `ng new`
2. `cd` into the root folder
3. Run ` npx nx@latest init --integrated` (or `npx nx@latest init`, also fails)


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 20.13.1
OS     : darwin-arm64
npm    : 10.5.2

nx (global)  : 19.1.1
nx           : 19.2.3
@nrwl/tao    : 19.2.3
```

### Failure Logs

```shell

```

### Package Manager Version

_No response_

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

I just tried something else in case it gives you more info: running an older version of `nx` also fails, but the error trace is a bit more informative:

`npx nx@19 init --integrated`

<details><summary>Details</summary>
<p>

alvaro@Alvaros-MacBook-Pro-2 nx-31291-repro % npx nx@19 init --integrated

 NX   🧐 Checking versions compatibility

nx init

Adds Nx to any type of workspace. It installs nx, creates an nx.json configuration file and optionally sets up remote caching. For more info, check https://nx.dev/recipes/adopting-nx.

Options:
  --help                  Show help                                                                                                                                                                                                                               [boolean]
  --version               Show version number                                                                                                                                                                                                                     [boolean]
  --nxCloud               Set up distributed caching with Nx Cloud.                                                                                                                                                                                               [boolean]
  --interactive           When false disables interactive input prompts for options.                                                                                                                                                              [boolean] [default: true]
  --useDotNxInstallation  Initialize an Nx workspace setup in the .nx directory of the current repository.                                                                                                                                       [boolean] [default: false]

Error: Cannot find module '@angular/core'
Require stack:
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/package-json.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/plugins/loader.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/plugins/index.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/run/executor-utils.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/devkit-internals.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/assert-workspace-validity.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/build-project-graph.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/project-graph.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/file-utils.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/package-manager.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/child-process.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/migrate/command-object.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/nx-commands.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/bin/init-local.js
- /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/bin/nx.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Function.resolve (node:internal/modules/helpers:190:19)
    at readModulePackageJson (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/package-json.js:155:36)
    at getLegacyMigrationFunctionIfApplicable (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/init/implementation/angular/legacy-angular-versions.js:27:69)
    at addNxToAngularCliRepo (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/init/implementation/angular/index.js:26:106)
    at Object.initHandler (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/init/init-v2.js:66:51)
    at Object.handler (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/init/command-object.js:14:40)
    at /Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/yargs/build/index.cjs:1:8993
    at j (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/yargs/build/index.cjs:1:4956)
    at _.handleValidationAndGetResult (/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/yargs/build/index.cjs:1:8962) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/package-json.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/plugins/loader.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/plugins/index.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/run/executor-utils.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/devkit-internals.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/assert-workspace-validity.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/build-project-graph.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/project-graph.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/project-graph/file-utils.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/package-manager.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/utils/child-process.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/migrate/command-object.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/src/command-line/nx-commands.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/bin/init-local.js',
    '/Users/alvaro/.npm/_npx/9b714189d7fd5670/node_modules/nx/bin/nx.js'
  ]
}

</p>
</details>

--- a/packages/nx/src/command-line/init/implementation/angular/legacy-angular-versions.ts
+++ b/packages/nx/src/command-line/init/implementation/angular/legacy-angular-versions.ts
@@ -2,6 +2,7 @@ import { execSync } from 'child_process';
 import { join } from 'path';
 import { gte, major } from 'semver';
 import { readJsonFile, writeJsonFile } from '../../../../utils/fileutils';
+import { getNxRequirePaths } from '../../../../utils/installation-directory';
 import { sortObjectByKeys } from '../../../../utils/object-sort';
 import { output } from '../../../../utils/output';
 import { readModulePackageJson } from '../../../../utils/package-json';
@@ -11,9 +12,9 @@ import {
   resolvePackageVersionUsingInstallation,
   resolvePackageVersionUsingRegistry,
 } from '../../../../utils/package-manager';
+import { connectExistingRepoToNxCloudPrompt } from '../../../connect/connect-to-nx-cloud';
 import { initCloud } from '../utils';
 import type { Options } from './types';
-import { connectExistingRepoToNxCloudPrompt } from '../../../connect/connect-to-nx-cloud';
 
 // map of Angular major versions to Nx versions to use for legacy `nx init` migrations,
 // key is major Angular version and value is Nx version to use
@@ -35,9 +36,25 @@ export async function getLegacyMigrationFunctionIfApplicable(
   repoRoot: string,
   options: Options
 ): Promise<() => Promise<void> | null> {
-  const angularVersion =
-    readModulePackageJson('@angular/core').packageJson.version;
-  const majorAngularVersion = major(angularVersion);
+  let majorAngularVersion: number;
+  try {
+    const angularVersion = readModulePackageJson('@angular/core', [
+      repoRoot,
+      ...getNxRequirePaths(),
+    ]).packageJson.version;
+    majorAngularVersion = major(angularVersion);
+  } catch {
+    output.error({
+      title: 'Could not determine the existing Angular version',
+      bodyLines: [
+        'Please ensure "@angular/core" is installed in the workspace and you are running this command from the root of the workspace.',
+      ],
+    });
+    // errors are caught in the initHandler and not logged to the user, so we
+    // log it first and then throw to ensure it is logged to the user
+    throw new Error('Could not determine the existing Angular version');
+  }
+
   if (majorAngularVersion >= minMajorAngularVersionSupported) {
     // non-legacy
     return null;

